blob/s3blob/s3blob.go

```diff
@@ -628,11 +663,33 @@ func unescapeKey(key string) string {
 // NewTypedWriter implements driver.NewTypedWriter.
 func (b *bucket) NewTypedWriter(ctx context.Context, key string, contentType string, opts *driver.WriterOptions) (driver.Writer, error) {
 	key = escapeKey(key)
-	uploader := s3manager.NewUploaderWithClient(b.client, func(u *s3manager.Uploader) {
-		if opts.BufferSize != 0 {
-			u.PartSize = int64(opts.BufferSize)
-		}
-	})
+
+	// Check if a cached uploader already exists.
```

Comment:
This would be more readable if it were factored out into a helper function, e.g. `func (b *bucket) uploader(opts *driver.WriterOptions) *s3manager.Uploader`.